code/handlers/trackservers.q

```diff
@@ -175,13 +173,21 @@ retrydiscovery:{
 // Called by the discovery service when it restarts
 autodiscovery:{if[DISCOVERYRETRY>0; .servers.retrydiscovery[]]}
 
+//check if .proc.getattributes exists in the nontorqprocess
+checknontorqattr:{(`proc in key`)&`getattributes in key`.proc};
+
 // Attempt to make a connection for specified row ids
 retryrows:{[rows]
+    //function a checks if the handle passed is empty and also invokes checknontorqattr function 
+    //which checks if .proc.getattributes is defined on the nontorqprocess and executes it 
+    //only if it is defined
+    a:{$[(first not null x)&@[first x;(checknontorqattr;`);0b];@[first x;(`.proc.getattributes;`);()!()];()!()]};
```

Comment:
I think this should maybe be a standalone function for checking attributes on a remote process

I thnk the error trap in first clause of `$` should also remove the need to check if `x` is null - if it is, error trap will catch it anyway.

Also if you're only passing one handle in, as the call below appears to do, should be no need for all these `first`s

Finally, according to our internal style suggestions, it is preferred to control execution order with square brackets rather than parentheses i.e.
```q
first[not null x]
```
rather than
```q
(first not null x)
```
(but as above, I think this part can actually be removed completely)